code/scripting/api/libs/mission.cpp

```diff
@@ -1510,6 +1510,26 @@ ADE_FUNC(getArrivalList,
 	                    l_ParseObject.Set(parse_object_h(&Ship_arrival_list)));
 }
 
+ADE_FUNC(getShipList,
+	l_Mission,
+	nullptr,
+	"Get an iterator to the list of ships in this mission",
+	"iterator<ship>",
+	"An iterator across all ships in the mission. Can be used in a for .. in loop")
+{
+	ship_obj* so = &Ship_obj_list;
+
+	return ade_set_args(L, "u", luacpp::LuaFunction::createFromStdFunction(L, [so](lua_State* LInner, const luacpp::LuaValueList& /*params*/) mutable -> luacpp::LuaValueList {
+		so = GET_NEXT(so);
```

Comment:
Add a comment explaining why `GET_FIRST` isn't used.  Initially I thought this was a bug, as all loops start by calling `GET_FIRST`, but then I realized this would work if the iterator iterated before returning the first object.

Also affirmatively indicate that this should work on empty lists, which I think is true for the same reason as above.